src/main/java/com/google/sps/servlets/GoDataServlet.java

```diff
@@ -42,7 +45,15 @@
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String indexStr = request.getParameter(INDEX_PARAMETER);
-    index = Integer.parseInt(indexStr);
+    try {
+      index = Integer.parseInt(indexStr);
+    } catch (Exception e) {
+    }
+
+    name = request.getParameter(NAME_PARAMETER);
```

Comment:
It's not clear that name is a guess, especially considering the new argument to the ScavengerHunt constructor is called "guess".

```suggestion
    nameGuess = request.getParameter(NAME_PARAMETER);
```